codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/EventSourcedEntityTestKitGenerator.scala

```diff
@@ -0,0 +1,189 @@
+import com.lightbend.akkasls.codegen.java.SourceGenerator._
+import com.lightbend.akkasls.codegen.java.EntityServiceSourceGenerator.generateImports
+import _root_.java.nio.file.{Files, Path}
+
+object EventSourcedEntityTestKitGenerator {
+
+  def generate(entity: ModelBuilder.Entity,
+               service: ModelBuilder.EntityService,
+               generatedSourceDirectory: Path): Iterable[Path] = {
+    val packageName = entity.fqn.parent.javaPackage
+    val className = entity.fqn.name
+    val sourceCode = generateSource(service, entity, packageName, className)
+
+    val packagePath = packageAsPath(packageName)
+    val generatedSourceDirectoryPath = generatedSourceDirectory.resolve(packagePath.resolve(className + "TestKit.java"))
+
+    if (!generatedSourceDirectoryPath.toFile.exists) {
+      Files.write(generatedSourceDirectoryPath, sourceCode.getBytes(Charsets.UTF_8))
+      List(generatedSourceDirectoryPath)
+    } else {
+      Nil
+    }
+
+  }
+
+  private[codegen] def generateSource(service: ModelBuilder.EntityService,
+                                      entity: ModelBuilder.Entity,
+                                      packageName: String,
+                                      className: String): String = {
+    entity match {
+      case entity: ModelBuilder.EventSourcedEntity => generateSourceCode(service, entity, packageName, className)
+      case entity: ModelBuilder.ValueEntity => "/** FIXME implement Value Entity testkit */"
+    }
+  }
+
+  private[codegen] def generateSourceCode(service: ModelBuilder.EntityService,
+                                          entity: ModelBuilder.EventSourcedEntity,
+                                          packageName: String,
+                                          className: String): String = {
+    val imports = generateImports(
+      service.commands,
+      entity.state,
+      packageName,
+      otherImports = Seq(
+        "com.google.protobuf.Empty",
+        "java.util.ArrayList",
+        "java.util.List",
+        "java.util.NoSuchElementException",
+        "scala.jdk.javaapi.CollectionConverters",
+        "com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntity",
+        "com.akkaserverless.javasdk.impl.effect.SecondaryEffectImpl",
+        "com.akkaserverless.javasdk.impl.effect.MessageReplyImpl",
+        "com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityEffectImpl",
+        "com.akkaserverless.javasdk.testkit.Result",
+        "com.akkaserverless.javasdk.testkit.internal.AkkaServerlessTestKitHelper"
+      )
+    )
+
+    val domainClassName = entity.fqn.parent.javaOuterClassname
+    val entityClassName = entity.fqn.name
+    val entityStateName = entity.state.fqn.name
+
+    val testkitClassName = s"${entityClassName}TestKit"
+
+    s"""$managedCodeCommentString
+          |package ${entity.fqn.parent.pkg};
+          |
+          |import ${service.fqn.parent.pkg}.${service.fqn.parent.javaOuterClassname};
```

Comment:
Do we really need this import? Normally, `generateImports` should be able to find out all the types that are need to be included.